Invalid sql in migration after relative path change

Got a lot of these after the relative path change. Postgres DB.

And the app doesnt even boot.

]
        ... 49 common frames omitted
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::populate-podcast-channel-media-file-id::anon:
     Reason: liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.CustomChangeException: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:124) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.lambda$null$0(Liquibase.java:265) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.lambda$update$1(Liquibase.java:264) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.runInScope(Liquibase.java:2404) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.update(Liquibase.java:211) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.update(Liquibase.java:197) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:314) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:269) ~[liquibase-core-4.6.2.jar:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[spring-beans-5.3.8.jar:5.3.8]
        ... 61 common frames omitted
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::populate-podcast-channel-media-file-id::anon:
     Reason: liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.CustomChangeException: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:695) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:49) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:110) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:244) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66) ~[liquibase-core-4.6.2.jar:na]
        ... 79 common frames omitted
Caused by: liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.CustomChangeException: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at liquibase.change.custom.CustomChangeWrapper.generateStatements(CustomChangeWrapper.java:169) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1259) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:660) ~[liquibase-core-4.6.2.jar:na]
        ... 99 common frames omitted
Caused by: liquibase.exception.CustomChangeException: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at org.airsonic.player.spring.migrations.AddMediaFileIdToPodcastChannels.generateStatements(AddMediaFileIdToPodcastChannels.java:75) ~[classes/:na]
        at liquibase.change.custom.CustomChangeWrapper.generateStatements(CustomChangeWrapper.java:162) ~[liquibase-core-4.6.2.jar:na]
        ... 101 common frames omitted
Caused by: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at liquibase.database.jvm.JdbcConnection.prepareStatement(JdbcConnection.java:389) ~[liquibase-core-4.6.2.jar:na]
        at org.airsonic.player.spring.migrations.AddMediaFileIdToPodcastChannels.generateStatements(AddMediaFileIdToPodcastChannels.java:66) ~[classes/:na]
        ... 102 common frames omitted
Caused by: org.postgresql.util.PSQLException: Unterminated string literal started at position 102 in SQL SELECT id FROM media_file WHERE path='/mnt/nas/common/Podcasts/Dan Carlin's Hardcore History- Addendum';. Expected  char
        at org.postgresql.core.Parser.checkParsePosition(Parser.java:1328) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.core.Parser.parseSql(Parser.java:1235) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.core.Parser.replaceProcessing(Parser.java:1179) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.core.CachedQueryCreateAction.create(CachedQueryCreateAction.java:43) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.core.CachedQueryCreateAction.create(CachedQueryCreateAction.java:19) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.util.LruCache.borrow(LruCache.java:123) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.core.QueryExecutorBase.borrowQuery(QueryExecutorBase.java:293) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.jdbc.PgConnection.borrowQuery(PgConnection.java:170) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.jdbc.PgPreparedStatement.<init>(PgPreparedStatement.java:83) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:1295) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:1735) ~[postgresql-42.2.22.jar:42.2.22]
        at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:428) ~[postgresql-42.2.22.jar:42.2.22]
        at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:337) ~[HikariCP-4.0.3.jar:na]
        at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java) ~[HikariCP-4.0.3.jar:na]
        at liquibase.database.jvm.JdbcConnection.prepareStatement(JdbcConnection.java:387) ~[liquibase-core-4.6.2.jar:na]

--- a/airsonic-main/src/main/java/org/airsonic/player/spring/migrations/AddMediaFileIdToPodcastChannels.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/spring/migrations/AddMediaFileIdToPodcastChannels.java
@@ -63,10 +63,12 @@ public class AddMediaFileIdToPodcastChannels implements CustomSqlChange {
                     String title = rs2.getString("title");
                     if (StringUtils.isNotEmpty(title)) {
                         Path channelFolder = folderPath.resolve(StringUtil.fileSystemSafe(title));
-                        try (PreparedStatement st3 = conn.prepareStatement("SELECT id FROM media_file WHERE path='" + channelFolder.toString() + "';");
-                                ResultSet rs3 = st3.executeQuery()) {
-                            while (rs3.next()) {
-                                idToMediaFile.put(rs2.getInt("id"), rs3.getInt("id"));
+                        try (PreparedStatement st3 = conn.prepareStatement("SELECT id FROM media_file WHERE path=?;")) {
+                            st3.setString(1, channelFolder.toString());
+                            try (ResultSet rs3 = st3.executeQuery()) {
+                                while (rs3.next()) {
+                                    idToMediaFile.put(rs2.getInt("id"), rs3.getInt("id"));
+                                }
                             }
                         }
                     }

